### ISSUE TITLE
10 second overhead for batch jobs

### ISSUE THREAD

**Issue by quantumjim** [Qiskit/qiskit-aer#1015]
### Informations

- **Qiskit Aer version**: 0.7.0
- **Python version**: 3.8.5
- **Operating system**: OSX Catalina

### What is the current behavior?

A single call to `execute` with a list of circuits takes around 10s longer than calling it for each circuit sequentially.

### Steps to reproduce the problem

Example code

```
from qiskit import execute, Aer, __qiskit_version__
from qiskit.circuit.random import random_circuit
import time
# make two random circuits
qcs = [random_circuit(2, 2, measure=True) for _ in range(2)]
# run them all at once and see how long it takes
t0 = time.time()
execute(qcs,Aer.get_backend('qasm_simulator'))
print('Time when batched',time.time()-t0)
# run them sequentially and see how long it takes
t0 = time.time()
for qc in qcs:
    execute(qc,Aer.get_backend('qasm_simulator'))
print('Time when run sequentially',time.time()-t0)
```

Similar results are found for different numbers 

### What is the expected behavior?

They should take around the same time. But when running the above I get the following

```
Time when batched 11.500309944152832
Time when run sequentially 0.00656580924987793
```

I do not see the same problem when running on IQX. There I instead get

```
Time when batched 0.6284997463226318
Time when run sequentially 0.2993638515472412
```

Which still isn't perfect, but it's better. So maybe it is an OSX problem?

### Suggested solutions

I should install Linux ;)

**Comment by mtreinish** [comment]
I expect this is actually overhead from `parallel_map` in terra for transpile. For python 3.8 multiprocessing on osx (especially 10.15) is problematic as 3.8 switched to spawn by default instead of fork. This causes both buggy operation and way more overhead. Can you try setting the env variable `QISKIT_IN_PARALLEL` to  `TRUE` and running? That will disable `parallel_map` in terra so the transpilation will be executed serially. Also you can try splitting the transpilation and assemble out from `backend.run` and see the overhead there. Basically do:

```python3
from qiskit import transpile
from qiskit import assemble

backend = Aer.get_backend('qasm_simulator')
qcs = [random_circuit(2, 2, measure=True) for _ in range(2)]
qobj = assemble([transpile(qc, backend) for qc in qcs], backend)
t0 = time.time()
backend.run(qobj)
print(time.time() - t0)
```

Or you could install Linux :)

**Comment by quantumjim** [comment]
Thanks. That certainly seems to speed up running the circuits, though assemble seems to have a bit of an overhead too.

**Comment by mtreinish** [comment]
Yeah, assemble also runs in `parallel_map` for assembling multiple circuits (it was added in 0.13.0 with Qiskit/qiskit-terra#3971). So it will suffer from the same issues with macOS.

 I'm going to transfer this issue to terra and push up a PR that will make parallel execution opt-in for macOS. It's a big performance hit for people on older versions of python and/or macOS, but I don't think I've had a single week go by since we added python 3.8 support that I've had to help someone fight these issues.
